### DISCUSSION TITLE
How to optimize trace of inverse in CVXPY?

### DISCUSSION THREAD

**Discussion by sebhan2** [cvxpy/cvxpy#1982] [General]
Hello,

how can I formulate an objective function that minimizes the trace of an inverse matrix using cvxpy? 

Concretely the following problem:

[![enter image description here][1]][1]

[![enter image description here][2]][2]

subject to:

[![enter image description here][3]][3]

  [1]: https://i.stack.imgur.com/souVx.png
  [2]: https://i.stack.imgur.com/VBkfp.png
  [3]: https://i.stack.imgur.com/KgWDw.png

I have seen the trace_inv objective function in cvx but so far was not able to figure out how to translate this to cvxpy.

I also posted my question to stackoverflow, but have not gotten an answer yet, here is a link: https://stackoverflow.com/questions/74882101/how-to-optimize-trace-of-inverse-in-cvxpy

**Comment by sebhan2** [comment]
Well, I figured it out by reading the source code. Here is my minimal example:

https://stackoverflow.com/questions/74882101/how-to-optimize-trace-of-inverse-in-cvxpy/74892871#74892871

**Comment by aze-539** [comment]
Hello, I also encountered a similar problem of finding trace of inverse. I would like to ask why **cp.Minimize(cp.matrix_frac(np.eye(6), cost))** can be solved, but when i try to solve **cp.Maximize(1/cp.real(cp.matrix_frac(np.eye(6), cost)))**, an error will be reported cvxpy.error.DCPError: Problem does not follow DCP rules. Specifically:
The objective is not DCP.

Shouldn't these two questions be equivalent?

**Reply by phschiele** [reply]
With CVXPY, you can minimize convex functions, and maximize concave functions. The first function is convex, but the reciprocal of a convex function is generally not concave.

**Comment by phschiele** [comment]
Closing this discussion, answered by author @sebhan2.
